### PR TITLE
Document how to work with third-party snaps

### LINKED ISSUE #760: Document how to connect to a snap, and how to find out whether a snap is installed
The JSON-RPC API in the docs shows `wallet_getSnaps` and `wallet_getSnaps`, but we should have more complete examples of each in the "How To" section of the docs.

### PR DESCRIPTION
**Pull request by ziad-saab**
PR adds a new page under the How To section that describes how to work with existing, third-party snaps.

Fixes #760

### CODE REVIEW

File: snaps/reference/rpc-api.md [hunk 1]
```diff
@@ -636,7 +636,7 @@ interact with the specified snaps.
 
 A website can also call this method to invoke the specified JSON-RPC method of the specified snap.
 
-This method is synonymous to [`wallet_invokeSnap`](#wallet_invokesnap), and is only callable by websites.
+This method is synonymous to [`wallet_invokeSnap`](#wallet_invokesnap).
```
Comment (Montoya) [inline snaps/reference/rpc-api.md:639]: Can we move this change into another PR? I want to approve this and hold off on the other part (I want to re-write the how-to).

**Review by Montoya — DISMISSED** [review]
I want to rewrite the how-to but I cannot do it right now.

File: snaps/how-to/work-with-existing-snaps.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+While some snaps are designed to be one-off and dedicated one and only one Dapp, this isn't always the case. Third-party developers like yourself can use these snaps to build your own Dapps, or even to compose with another snap.
+
+This is possible because [snaps can expose an RPC API](../reference/exports/#onrpcrequest). Snaps can decide to make their API available to other Dapps, other snaps, or both, by requesting the [`endowment:rpc`](../reference/permissions.md#endowmentrpc) permission.
+
+## Installing a snap
+
+Installing a snap is done by using the [`wallet_requestSnaps`](../reference/rpc-api.md#wallet_requestsnaps) method from your Dapp or your snap.
+
+:::tip
+A snap is not installed only once. It has to be installed for every consumer that wants to use it. For example, if you're building two Dapps that use the same snap, each Dapp will need to install the snap independently.
```
Comment (GuillaumeRx) [inline snaps/how-to/work-with-existing-snaps.md:12]: The snap is only installed one time when the first dApp requests the snap. All the other dApp that wants to use the snap will just need to request permission to talk to the snap

File: snaps/how-to/work-with-existing-snaps.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Working with third-party snaps
+
+While some snaps are designed to be one-off and dedicated one and only one Dapp, this isn't always the case. Third-party developers like yourself can use these snaps to build your own Dapps, or even to compose with another snap.
+
+This is possible because [snaps can expose an RPC API](../reference/exports/#onrpcrequest). Snaps can decide to make their API available to other Dapps, other snaps, or both, by requesting the [`endowment:rpc`](../reference/permissions.md#endowmentrpc) permission.
+
+## Installing a snap
+
+Installing a snap is done by using the [`wallet_requestSnaps`](../reference/rpc-api.md#wallet_requestsnaps) method from your Dapp or your snap.
```
Comment (GuillaumeRx) [inline snaps/how-to/work-with-existing-snaps.md:9]: We shouldn't endorse snaps talking or installing other snaps for the moment

Comment (FrederikBolding) [reply snaps/how-to/work-with-existing-snaps.md:9]: `wallet_requestSnaps` is blocked in snaps currently FYI.

File: snaps/how-to/work-with-existing-snaps.md (hunk 2 above)
Comment (eriknson) [reply snaps/how-to/work-with-existing-snaps.md:12]: I think we decided to use `website` instead of `dapp`, but could be wrong.

cc @coreyjanssen and maybe @rekmarks

File: snaps/how-to/work-with-existing-snaps.md (hunk 3 above)
Comment (ziad-saab) [reply snaps/how-to/work-with-existing-snaps.md:9]: @FrederikBolding I'm curious what's the reason that it's blocked? and how does that relate to the `snaps: true` in the `endowment:rpc`?

File: snaps/how-to/work-with-existing-snaps.md (hunk 2 above)
Comment (coreyjanssen) [reply snaps/how-to/work-with-existing-snaps.md:12]: You're correct, @eriknson! We've opted to say `site` instead of `dapp`, as it's more user friendly.

File: snaps/how-to/work-with-existing-snaps.md [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Connecting to a snap is done by using the [`wallet_requestSnaps`](../reference/rpc-api.md#wallet_requestsnaps) method from your Dapp. If a user does not have a snap installed in their MetaMask, MetaMask will prompt the user to install the snap. There are different possible outcomes from calling `wallet_requestSnaps`:
+
+
+### User rejects the install request
+
+If the user rejects the install request, the call to `wallet_requestSnaps` will throw with:
+
+```json
+{ code: 4001, message: "User rejected the request." Z}
```
Comment (Montoya) [inline snaps/how-to/work-with-existing-snaps.md:15]: Looks like a trailing character Z here

**Review by Montoya — CHANGES_REQUESTED** [review]

File: snaps/how-to/work-with-existing-snaps.md (hunk 4 above)
Comment (ziad-saab) [reply snaps/how-to/work-with-existing-snaps.md]: Resolved

**Review by Montoya — APPROVED** [review]

**Review by alexandratran — APPROVED** [review]
Made some content edits for style and clarity. LGTM.

File: snaps/how-to/work-with-existing-snaps.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+If a user doesn't have the snap installed in their MetaMask wallet, MetaMask prompts the user to
+install the snap.
+The following are different possible outcomes from calling `wallet_requestSnaps`.
+
+### User rejects the installation request
+
+If the user rejects the installation request, the call to `wallet_requestSnaps` throws the following error:
+
+```json
+{ code: 4001, message: "User rejected the request." }
```
Comment (Mrtenz) [inline snaps/how-to/work-with-existing-snaps.md:27]:
```suggestion
{ "code": 4001, "message": "User rejected the request." }
```

**Review by Montoya — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Review by Mrtenz — APPROVED** [review]

**Review by Montoya — APPROVED** [review]

**Comment by alidarkok617** [comment]
I apologize, I just learned, if I made a mistake. I'm sorry

در تاریخ جمعه ۷ ژوئیه ۲۰۲۳،‏ ۷:۳۹ ب.ظ Ziad Saab ***@***.***>
نوشت:
